datafusion/src/optimizer/constant_folding.rs

```diff
@@ -210,6 +210,74 @@ impl<'a> ExprRewriter for Simplifier<'a> {
                         right,
                     },
                 },
+                Operator::Or => match (left.as_ref(), right.as_ref()) {
+                    (
+                        Expr::Literal(ScalarValue::Boolean(l)),
+                        Expr::Literal(ScalarValue::Boolean(r)),
+                    ) => match (l, r) {
+                        (Some(l), Some(r)) => {
+                            Expr::Literal(ScalarValue::Boolean(Some(*l || *r)))
+                        }
+                        _ => Expr::Literal(ScalarValue::Boolean(None)),
+                    },
+                    (Expr::Literal(ScalarValue::Boolean(b)), _)
+                        if self.is_boolean_type(&right) =>
```

Comment:
@alamb to help me understand, can you provide color on whats an example of an `Expr` that isnt a `Boolean` but that has a boolean type?

Reply:
nvm, i get it now.

Reply:
One example is if someone writes `col_a OR true` and `col_a` has type of `int64` or something -- I don't know how far such an expression will get (it will likely error out when trying to to actually run the `PhysicalExpr`)